### PR TITLE
refactor(schedule): separate the schedule module interface

### PR DESCRIPTION
**Pull request by guowl3**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:
separate the schedule module interface and adapt the existing code accordingly
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/model/CloudConnectionConfig.java [hunk 1]
```diff
@@ -15,13 +15,17 @@
...
+import java.util.Map;
+
 public interface CloudConnectionConfig {
     String getClusterName();
 
     String getTenantName();
 
     String getOBTenantName();
 
+    Map<String, String> getProperties();
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/connection/model/CloudConnectionConfig.java:27]: what's this used for?

you may take a look at `ConnectionConfig`, there is a field named `attributes`.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/model/TestConnectionReq.java [hunk 2]
```diff
@@ -141,6 +141,8 @@ public class TestConnectionReq implements CloudConnectionConfig, SSLConnectionCo
     @JsonIgnore
     private OBInstanceRoleType instanceRoleType;
 
+    private Map<String, String> properties;
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/connection/model/TestConnectionReq.java:144]: same as above

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DBTableController.java [hunk 3]
```diff
@@ -59,7 +61,12 @@ public ListResponse<String> listTables(@PathVariable String sessionId,
             @PathVariable(required = false) String databaseName,
             @RequestParam(required = false, name = "fuzzyTableName") String fuzzyTableName) {
         ConnectionSession session = sessionService.nullSafeGet(sessionId, true);
-        return Responses.list(tableService.showTablesLike(session, databaseName, fuzzyTableName));
+        if (!StringUtils.isEmpty(fuzzyTableName)) {
```
Comment (yizhouxw) [inline server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DBTableController.java:64]: empty fuzzyTableName has been handled in AchemaAccessor already ?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/utils/DataArchiveConditionUtil.java [hunk 4]
```diff
@@ -60,55 +62,68 @@ private static String replaceVariables(String condition, Map<String, String> var
...
     }
 
-    private static String calculateDateTime(Date baseDate, String pattern) {
-        String[] parts = pattern.split("\\|");
-        String offsetString = parts[1].substring(1);
-        ChronoUnit unit = parseUnit(offsetString);
-        long offsetSeconds = parseValue(offsetString) * unit.getDuration().getSeconds();
-        if (parts[1].startsWith("-")) {
-            offsetSeconds = -offsetSeconds;
+    private static String calculateDateTime(Date baseDate, OffsetConfig config) {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/utils/DataArchiveConditionUtil.java:79]: please add related unit testing cases

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/QuartzJobChangeReq.java [hunk 5]
```diff
@@ -15,44 +15,33 @@
...
- * @Date: 2023/5/24 15:35
+ * @Date: 2024/6/9 22:19
  * @Descripition:
  */
 
 @Data
-public class ScheduleTaskResp {
-
-    private Long id;
+public class QuartzJobChangeReq {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/QuartzJobChangeReq.java:31]: ChangeQuartJobReq


and, will it be used for front-end request , if yes, avoid quartz concept?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleService.java [hunk 6]
```diff
@@ -161,33 +128,113 @@ public class ScheduleService {
...
+        if (approvalFlowInstanceId == null) {
+            executeChangeSchedule(req);
         }
-        quartzJobService.createJob(buildCreateJobReq(scheduleConfig));
-        scheduleConfig.setStatus(ScheduleStatus.ENABLED);
-        scheduleRepository.save(scheduleConfig);
+
+        // create change log for this request
+        ScheduleChangeLog changeLog = scheduleChangeLogService.createChangeLog(
+                ScheduleChangeLog.build(targetSchedule.getId(), req.getOperationType(),
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleService.java:184]: dont use default value

File: server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/utils/DataArchiveConditionUtil.java (hunk 4 above)
Comment (guowl3) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/utils/DataArchiveConditionUtil.java:79]: done

**Review by yizhouxw — APPROVED** [review]

**Pull request merged**